File: service/GoodNight.Service.Domain/Write/Story.cs
using System;
using System.Collections.Immutable;

namespace GoodNight.Service.Domain.Write
{
  public record Story(
    string Name,
    IImmutableSet<Scene> Scenes,
    IImmutableSet<Quality> Qualities)
  {
    public string Urlname
    {
      get
      {
        return NameConverter.OfString(Name);
      }
    }


    // Create a new story with a given name.
    public static Story Create(string name)
    {
      return new Story(name,
        ImmutableHashSet<Scene>.Empty,
        ImmutableHashSet<Quality>.Empty);
    }
  }
}

File: client/src/model/write/Story.ts
import type {Scene} from "./Scene";
import type {Quality} from "./Quality";


export interface Category {
  name: string;
  categories: Array<Category>;
  scenes: Array<Scene>;
  qualities: Array<Quality>;
}


export interface Story {
  name: string;
  urlname: string;
  icon: string | null;
  description: string;
}

